Dockerfile

```diff
@@ -3,13 +3,18 @@ LABEL maintainer "Fco. Javier Delgado del Hoyo <frandelhoyo@gmail.com>"
 
 COPY ["run.sh", "backup.sh", "restore.sh", "/"]
 
-RUN apk add --update bash mysql-client gzip && rm -rf /var/cache/apk/* && mkdir /backup &&\
+RUN apk add --update bash mysql-client gzip openssl && rm -rf /var/cache/apk/* && mkdir /backup &&\
   chmod u+x /backup.sh /restore.sh
 
+ENV DOCKERIZE_VERSION v0.6.1
+RUN wget https://github.com/jwilder/dockerize/releases/download/$DOCKERIZE_VERSION/dockerize-alpine-linux-amd64-$DOCKERIZE_VERSION.tar.gz \
+    && tar -C /usr/local/bin -xzvf dockerize-alpine-linux-amd64-$DOCKERIZE_VERSION.tar.gz \
+    && rm dockerize-alpine-linux-amd64-$DOCKERIZE_VERSION.tar.gz
+
 ENV CRON_TIME="0 3 * * sun" \
     MYSQL_HOST="mysql" \
     MYSQL_PORT="3306"
 
 VOLUME ["/backup"]
 
-CMD ["/run.sh"]
+CMD dockerize -wait tcp://${MYSQL_HOST}:3306 /run.sh
```

Comment:
I tried that syntax in my first attempt and `dockerize` was not recognized. 

When using the exec form, the command is run without a shell. As dockerized is a `go` script, I'm not sure if it can be used without a shell. The `dockerize` [documentation](https://github.com/jwilder/dockerize#usage) use the shell form, not the exec.

Reply:
Ok. Two more questions:

- Have you checked if the environment variable MYSQL_HOST takes the value defined outside the Dockerfile?
- Can you use MYSQL_PORT to specify the port?

Thanks!

Reply:
> Have you checked if the environment variable MYSQL_HOST takes the value defined outside the Dockerfile?

Yes, I've deployed my compose using this modification. In my case, MYSQL_HOST takes the name of the service providing the database

> Can you use MYSQL_PORT to specify the port?

Fair point, I'll add it.

Reply:
I'm sorry for the mess of commits. I had to use a different computer for the last commit and things got messed when I had to merge from master...

However, the changes are the same. I've added a parameter `TIMEOUT` used by `dockerize`. This parameter indicates when to give up if the database is not ready.